### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by Synesso**
Hi Heiko. Just an update to the dependencies. This will get rid of the "possible binary incompatibility" warnings in one of my current projects.

Also included is the dependency update checking plugin.

### CODE REVIEW

**Comment by plokhotnyuk** [comment]
Heiko, please also release it after the merge...

Jsoniter-scala evolved a lot since v0.27.1, there was binary incompatible changes and change of artifact names: https://github.com/plokhotnyuk/jsoniter-scala/compare/v0.27.1...v0.29.20

File: project/plugins.sbt [hunk 1]
```diff
@@ -4,5 +4,6 @@ addSbtPlugin("com.geirsson"      % "sbt-scalafmt" % "1.4.0")
 addSbtPlugin("com.typesafe.sbt"  % "sbt-git"      % "0.9.3")
 addSbtPlugin("de.heikoseeberger" % "sbt-header"   % "4.1.0")
 addSbtPlugin("org.foundweekends" % "sbt-bintray"  % "0.5.3")
+addSbtPlugin("com.timushev.sbt"  % "sbt-updates"  % "0.3.4")
```
Comment (hseeberger) [inline project/plugins.sbt]: Please remove this line, I use it solely as a global plugin (until there will be CI automated releases, if that will ever happen).

Comment (Synesso) [reply project/plugins.sbt]: Done.

**Comment by plokhotnyuk** [comment]
Something wrong happened...

v1.21.1 is tagged but artifacts aren't available on the Maven Central:
http://repo1.maven.org/maven2/de/heikoseeberger/akka-http-argonaut_2.12/
http://repo1.maven.org/maven2/de/heikoseeberger/akka-http-avro4s_2.12/
http://repo1.maven.org/maven2/de/heikoseeberger/akka-http-circe_2.12/
http://repo1.maven.org/maven2/de/heikoseeberger/akka-http-jackson_2.12/
http://repo1.maven.org/maven2/de/heikoseeberger/akka-http-json4s_2.12/
http://repo1.maven.org/maven2/de/heikoseeberger/akka-http-jsoniter-scala_2.12/
http://repo1.maven.org/maven2/de/heikoseeberger/akka-http-play-json_2.12/
http://repo1.maven.org/maven2/de/heikoseeberger/akka-http-upickle_2.12/

The same is with Scala 2.11.x artifacts :(

**Pull request merged**